test/deps/panic-checker

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+# Looks for a panic by grepping for stacktraces in a log file.
+# If a panic is found, it exits with code 1 and prints the stacktrace plus the last log message.
+# Only the first panic is returned.
+# If no panics are found it exits with code 0
+#
+# NOTE: When a panic occurs in LXD at runtime via a mux handler, it is logged at info level because the net/http library
+# has a built-in recover. We are not handling panic recovery manually. Because it is logged at info level, this checker
+# will only find panics if the test suite is run with LXD_VERBOSE=1 or LXD_DEBUG=1.
+with open(sys.argv[1]) as file:
+    output = ""
+    lastline = ""
+    for line in file:
+        if output == "" and not re.search("^goroutine\s+\d+\s+\[running]:", line):
+            # Output empty and no regex match so nothing found yet.
+            lastline = line
+            continue
+
+        if lastline != "":
+            # Got a regex match on this line, start the output with the last line and set it empty.
+            output = lastline + line
+            lastline = ""
+            continue
+
+        if re.search("(INFO|DEBUG|TRACE|WARNING|ERROR)", line):
```

Comment:
Just to be safe, maybe also check here if we reached EOF in case the panic is the last log message in the file.

Reply:
I guess this is just when the `for line in file` loop exits. I'll add a check if there is any output at that point and print it.